src/bentoml/_internal/frameworks/diffusers.py

```diff
@@ -0,0 +1,355 @@
+from __future__ import annotations
+
+import os
+import shutil
+import typing as t
+import logging
+from typing import TYPE_CHECKING
+
+import attr
+
+import bentoml
+from bentoml import Tag
+from bentoml.models import ModelContext
+from bentoml.exceptions import NotFound
+from bentoml.exceptions import BentoMLException
+from bentoml.exceptions import MissingDependencyException
+
+from ..models.model import PartialKwargsModelOptions
+
+if TYPE_CHECKING:
+    from types import ModuleType
+
+    from bentoml.types import ModelSignature
+    from bentoml.types import ModelSignatureDict
+
+
+try:
+    import torch
+    import diffusers
+    from diffusers.utils.import_utils import is_torch_version
+    from diffusers.utils.import_utils import is_xformers_available
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        "'diffusers' is required in order to use module 'bentoml.diffusers', install diffusers with 'pip install --upgrade diffusers transformers accelerate'. For more information, refer to https://github.com/huggingface/diffusers",
+    )
+
+
+MODULE_NAME = "bentoml.diffusers"
+DIFFUSION_MODEL_FOLDER = "diffusion_model"
+DIFFUSION_MODEL_CONFIG_FILE = "model_index.json"
+API_VERSION = "v1"
+
+logger = logging.getLogger(__name__)
+
+
+@attr.define
+class DiffusersOptions(PartialKwargsModelOptions):
+    """Options for the diffusers model."""
+
+    pipeline_class: type[diffusers.pipelines.DiffusionPipeline] | None = None
+    scheduler_class: type[diffusers.SchedulerMixin] | None = None
```

Comment:
As discussed, update the type to `str`.

Reply:
I have a discussion with aaron later and we agree that diffusers' situation is different from transformers, so currently we may keep using the class type instead of str. This is also eaiser for user to use their own pipelines.